### ISSUE TITLE
Helm ignoring CLI passed `--registry-config` if `HELM_CONFIG_HOME` isn't defined

### ISSUE THREAD

**Issue by kuzaxak** [helm/helm#11345]
Output of `helm version`:
```
version.BuildInfo{Version:"v3.9.4", GitCommit:"dbc6d8e20fe1d58d50e6ed30f09a04a77e4c68db", GitTreeState:"clean", GoVersion:"go1.17.13"}
```

Output of `kubectl version`:
```
bash: kubectl: command not found
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

`AWS EKS`

Bug report:

Helm ignoring CLI passed registry config file if `HELM_CONFIG_HOME` isn't defined.
I have a config located at path `/tmp/helm/registry/config.json` and it contains auth only a setting for ECR OCI. If I define `HELM_CONFIG_HOME` to `/tmp/helm` all command works fine. But If I unset it or move a file to custom location helm fails to authorize a request:

```
ubuntu@r-test-2:/usr/src/app$ helm --registry-config /tmp/helm/registry/config.json --debug show chart oci://662491802882.dkr.ecr.us-east-1.amazonaws.com/glia-application
show.go:195: [debug] Original chart version: ""
apiVersion: v2
description: Shared Helm library chart for Glia Applications
name: glia-application
type: application
version: 2.17.0

ubuntu@r-test-2:/usr/src/app$ unset HELM_CONFIG_HOME
ubuntu@r-test-2:/usr/src/app$ helm --registry-config /tmp/helm/registry/config.json --debug show chart oci://662491802882.dkr.ecr.us-east-1.amazonaws.com/glia-application
show.go:195: [debug] Original chart version: ""
Error: pulling from host 662491802882.dkr.ecr.us-east-1.amazonaws.com failed with status code [manifests 2.17.0]: 401 Unauthorized
helm.go:84: [debug] pulling from host 662491802882.dkr.ecr.us-east-1.amazonaws.com failed with status code [manifests 2.17.0]: 401 Unauthorized
```

**Comment by kuzaxak** [comment]
Can reproduce it on `3.9.4` but not on `3.10.0-rc.1`.

**Comment by joejulian** [comment]
This was fixed by #10762 and will be released in 3.10.0

**Comment by kuzaxak** [comment]
> This was fixed by #10762 and will be released in 3.10.0

Thank you for pointing me to the correct PR!
